spark/src/main/scala/org/apache/spark/sql/delta/SnapshotManagement.scala

```diff
@@ -498,30 +490,20 @@ trait SnapshotManagement { self: DeltaLog =>
    * file as a hint on where to start listing the transaction log directory. If the _delta_log
    * directory doesn't exist, this method will return an `InitialSnapshot`.
    */
-  protected def getSnapshotAtInit: CapturedSnapshot = {
+  protected def getSnapshotAtInit: CapturedSnapshot = withSnapshotLockInterruptibly {
```

Comment:
To make sure I'm understanding the new flow correctly:
* At construction time, `getSnapshotAtInit`:
  * calls `createLogSegment` with last checkpoint (if any) as the starting point
  * passes the resulting log segment to `getUpdatedSnapshot` (which handles commit owner changes)
  * directly returns the resulting `CapturedSnapshot` so it can be assigned
* At update time, `updateInternal`:
  * calls `createLogSegment` with current snapshot as the starting point
  * passes the resulting log segment to `getUpdatedSnapshot` (which handles commit owner changes)
  * calls `installSnapshot` to update or replace the captured snapshot as needed.

Nice. Much cleaner than the previous flow, which was creating log segments in different ways for init vs. update paths. Good-bye, `createSnapshotAtInitInternal`!

Reply:
nicely summarized.